src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/TracerProviderBuilderExtensions.cs

```diff
@@ -42,7 +43,10 @@ public static class TracerProviderBuilderExtensions
             var elasticsearchClientOptions = new ElasticsearchClientInstrumentationOptions();
             configure?.Invoke(elasticsearchClientOptions);
 
-            builder.AddInstrumentation((activitySource) => new ElasticsearchClientInstrumentation(activitySource, elasticsearchClientOptions));
+            builder.AddInstrumentation(() => new ElasticsearchClientInstrumentation(elasticsearchClientOptions));
+            builder.AddSource(ElasticsearchActivitySourceHelper.ActivitySourceName);
+            builder.AddLegacyActivity("Elasticsearch.Net.RequestPipeline.CallElasticsearch");
```

Comment:
@cijothomas updated to use `AddLegacyActivity`. Am I doing it correctly?

Reply:
Looks correct. @utpilla and myself was planning to modify one of the instrumentations in -contrib to leverage the new ActivitySourceAdapter-free verion. Glad to see you are ahead of us!

@utpilla Please take a look. I'll do a detailed review Monday.

Reply:
This looks good!

Reply:
ok. We can modify the instrumentation unit test to validate that the activity has IsAllDataRequested/Sampled populated correctly based on the sampler. This will validate that those activities are fed through Samplers.

Reply:
Ok, I will give that a try.

Reply:
@cijothomas I added a sampler test and it does not appear to be working correctly. I think I may need `SuppressInstrumentationScope.IncrementIfTriggered` and `SuppressInstrumentationScope.DecrementIfTriggered` to be public so that their calls can be added back into `DiagnosticSourceListener`. I had to remove it from [here ](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/DiagnosticSourceInstrumentation/DiagnosticSourceListener.cs#L56) since it's currently internal.

Reply:
I left couple of comments in the test code. If it still fails, can you verify if the activity.OperationName for the activity created by ES client is "Elasticsearch.Net.RequestPipeline.CallElasticsearch" itself? Not sure if its case sensitive.

Reply:
Are we still thinking about renaming to `AddLegacySource`?

Reply:
@CodeBlanch Yeah we had decided to update the name of the extension method to `AddLegacySource` in a different PR after this [one](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1836).

@ejsmith You have to pass `CallElasticsearch` to `AddLegacyActivity` as that's the OperationName of the activity created by the DiagnosticListener. You'll also have to update the method name to `AddLegacySource` when it gets updated in the new package.

Reply:
Thanks @utpilla that worked!